### PR TITLE
User not signed in on refresh

### PR DESCRIPTION
**Pull request by Simpcyclassy**

### CODE REVIEW

File: src/components/app/reducers.js [hunk 1]
```diff
@@ -26,6 +32,15 @@ export default ( state = { ...initialState }, action) => {
         };
     }
 
+    case CHECK_USER_LOGIN_STATUS: {
+        const user = Cookies.get(state.token);        
+        
+        return {
+            ...state,
+            isLoggedIn: !!user === true,
```
Comment (Way2nnadi) [inline src/components/app/reducers.js]: all you need is `!!user`

File: src/components/app/reducers.js [hunk 2]
```diff
@@ -15,7 +21,7 @@ export default ( state = { ...initialState }, action) => {
     switch (action.type) {
     case SET_COOKIE:    {
         const { data: { token, user }, status } = action.payload;
-        Cookies.set(token, { expires: EXPIRATION });
+        Cookies.set( token, user, { expires: EXPIRATION });
```
Comment (Way2nnadi) [inline src/components/app/reducers.js]: remove space before `token`

**Comment by Way2nnadi** [comment]
@chioma can you upload of video showing you testing this functionality?

**Comment by Simpcyclassy** [comment]
Sure, only that I have to test it with the token as a unique string cause the token being sent is not unique which is not meant to be so.

**Comment by Simpcyclassy** [comment]
ezgif com-video-to-gif

File: src/components/app/actions.js [hunk 3]
```diff
@@ -12,6 +18,18 @@ export const setCookie = payload => ({
...
 
+/**
+ * Triggers request to set cookie
+ *
+ * @function
+ * @params {String} payload - The data passed into the the action
+ * @return {Object} The {@link actionTypes.CHECK_USER_LOGIN_STATUS CHECK_USER_LOGIN_STATUS} action.
+ */
+export const checkUserLoginStatus = payload => ({
+    payload,
```
Comment (EngrEric) [inline src/components/app/actions.js]: you aren't making use of the payload, remove then

**Comment by EngrEric** [comment]
looks good but your bug is not fixed yet, i ran the branch locally but it seems not fixed.

**Review by EngrEric — CHANGES_REQUESTED** [review]
Looks good but needs to retain user data on refresh too.
It does not work for now

**Review by joshuaobinna — COMMENTED** [review]
Looks good but more work is needed

File: package.json [hunk 4]
```diff
@@ -1,6 +1,10 @@
 {
   "name": "agroeasy",
+<<<<<<< HEAD
+  "version": "1.2.18",
+=======
   "version": "1.3.0",
+>>>>>>> master
```
Comment (enyeInc) [inline package.json]: you messed up the merge conflicts

File: src/components/app/components/App.js [hunk 5]
```diff
@@ -10,13 +10,20 @@ import Footer from './Footer';
...
 const { ContactUs } = contactus.components;
 const { Content } = Layout;
 
 class App extends React.Component {
 
+    componentDidMount(){
+        const { actions: { checkUserLoginStatus } } = this.props;
+
+        checkUserLoginStatus();
+
```
Comment (enyeInc) [inline src/components/app/components/App.js:24]: nit - remove empty line

File: src/components/app/reducers.js [hunk 6]
```diff
@@ -13,9 +19,9 @@ const initialState = {
 export default ( state = { ...initialState }, action) => {
     
     switch (action.type) {
-    case SET_COOKIE:    {
+    case SET_COOKIE: {
         const { data: { token, user }, status } = action.payload;
-        Cookies.set(token, { expires: EXPIRATION });
+        Cookies.set("token", user, { expires: EXPIRATION });
```
Comment (enyeInc) [inline src/components/app/reducers.js:24]: `"token"` should be saved to the constants file

File: src/components/app/reducers.js [hunk 7]
```diff
@@ -26,10 +32,19 @@ export default ( state = { ...initialState }, action) => {
...
+        
+        return {
+            ...state,
+            isLoggedIn: !!user,
+        };
+    }
+
     case REMOVE_COOKIE: {
-        Cookies.expire(state.token);
+        Cookies.expire("token");
```
Comment (enyeInc) [inline src/components/app/reducers.js:45]: here as well

File: package.json (hunk 4 above)
Comment (Simpcyclassy) [reply package.json]: wow. I did not save correctly

**Comment by EngrEric** [comment]
i will be closing this PR given that we are adopting a new method for sign in

**Pull request closed without merging**